src/openssl/engine.c

```diff
@@ -297,9 +300,8 @@ static int by_subj_old_hash(X509_LOOKUP *lu, X509_LOOKUP_TYPE t, const X509_NAME
     if (count == 0) return 0;
 
     X509_STORE *store = X509_LOOKUP_get_store(lu);
-    STACK_OF(X509_OBJECT) *objs = X509_STORE_get1_objects(store);
+    STACK_OF(X509_OBJECT) *objs = X509_STORE_get0_objects(store);
```

Comment:
I see that the `get0` variant returns an internal pointer which must not be free'd by the caller, whereas the `get1` variant is a copy (of a list) that needs to be freed. It isn't clear from the openssl docs if the objects themselves are also internal (`get0`) or copied (`get1`).

what I'm getting at is are you comfortable that there are no refcount discrepancies with the x509 object `obj`?

Reply:
I believe so, x509 refcount is bumped when it is set on `obj`